File: _system_scripts/consolidate_folders.py
import os
import shutil

ROOT_DIR = r"c:\Users\hirak\Desktop\2nd-Brain\18_レミ投資漫画"
ARCHIVE_DIR = os.path.join(ROOT_DIR, "99_知識保管庫")

# Folders to move to Archive
TARGET_FOLDERS = [
    "01_投資の基礎知識",
    "02_マインド・哲学",
    "03_戦略・リスク管理",
    "04_未来・テクノロジー"
]

def main():
    if not os.path.exists(ARCHIVE_DIR):
        os.makedirs(ARCHIVE_DIR)
        print(f"Created archive directory: {ARCHIVE_DIR}")
        
    for folder_name in TARGET_FOLDERS:
        src_path = os.path.join(ROOT_DIR, folder_name)
        dest_path = os.path.join(ARCHIVE_DIR, folder_name)
        
        if os.path.exists(src_path):
            try:
                # If destination already exists, we might have a conflict.
                # shutil.move will move the folder INSIDE if dest exists and is a dir, 
                # or rename if it doesn't. We want to be careful.
                # Best to ensure dest parent exists (it does) and dest itself does NOT exist?
                # Or just merge?
                
                # If dest archive folder exists, we should probably merge or rename contents.
                # simple move:
                if os.path.exists(dest_path):
                    print(f"Destination {dest_path} already exists. Merging contents...")
                    # Allow move to merge? No, shutil.move might fail or nest.
                    # Let's use a function to move contents if dir exists.
                    for item in os.listdir(src_path):
                        s = os.path.join(src_path, item)
                        d = os.path.join(dest_path, item)
                        if os.path.exists(d):
                            print(f"  Skipping {item} (already in archive)")
                        else:
                            shutil.move(s, d)
                    # Try to remove empty src dir
                    try:
                        os.rmdir(src_path)
                        print(f"Moved contents and removed source: {folder_name}")
                    except OSError:
                        print(f"Moved contents but source not empty: {folder_name}")
                else:
                    shutil.move(src_path, dest_path)
                    print(f"Moved: {folder_name} -> 99_知識保管庫")
                    
            except Exception as e:
                print(f"Error moving {folder_name}: {e}")
        else:
            print(f"Not found (skipping): {folder_name}")

if __name__ == "__main__":
    main()

File: _system_scripts/old_tools/consolidate_folders.js
const fs = require('fs');
const path = require('path');

const ROOT_DIR = "c:\\Users\\hirak\\Desktop\\2nd-Brain\\18_レミ投資漫画";
const ARCHIVE_DIR = path.join(ROOT_DIR, "99_知識保管庫");
const TARGET_FOLDERS = [
    "01_投資の基礎知識",
    "02_マインド・哲学",
    "03_戦略・リスク管理",
    "04_未来・テクノロジー"
];

function moveFolder(src, dest) {
    if (!fs.existsSync(src)) {
        console.log(`Not found: ${src}`);
        return;
    }

    // If destination folder doesn't exist, just rename src to dest.
    if (!fs.existsSync(dest)) {
        try {
            fs.renameSync(src, dest);
            console.log(`Moved: ${path.basename(src)} -> 99_知識保管庫`);
        } catch (e) {
            console.error(`Error moving ${src}: ${e.message}`);
        }
        return;
    }

    // If destination DOES exist, we need to move contents.
    console.log(`Merging contents of ${src} into ${dest}`);
    const files = fs.readdirSync(src);
    for (const file of files) {
        const s = path.join(src, file);
        const d = path.join(dest, file);

        if (fs.existsSync(d)) {
            console.log(`Skipping duplicate: ${file}`);
        } else {
            try {
                fs.renameSync(s, d);
            } catch (e) {
                console.error(`Error moving file ${file}: ${e.message}`);
            }
        }
    }

    // Remove empty src dir
    try {
        fs.rmdirSync(src);
        console.log(`Removed empty source: ${src}`);
    } catch (e) {
        console.error(`Could not remove source (not empty?): ${e.message}`);
    }
}

function main() {
    if (!fs.existsSync(ARCHIVE_DIR)) {
        fs.mkdirSync(ARCHIVE_DIR);
        console.log("Created archive directory.");
    }

    for (const folder of TARGET_FOLDERS) {
        moveFolder(path.join(ROOT_DIR, folder), path.join(ARCHIVE_DIR, folder));
    }
}

main();
